src/utils/occurrence.ts

```diff
@@ -1,37 +1,30 @@
+import {Term} from "./term";
 export class Occurrence {
-    public term : string;
+    public term : Term;
     public positions : Array<number>;
 
-    constructor(t : string, p : Array<number>) {
+    constructor(t : Term, p : Array<number>) {
         this.term = t;
         this.positions = p;
     };
 
 
-    /**
-     * Transforms a map of [term] => indexes[] into an array of occurrences
-     * @param occurrenceMap                             map of [term] => indexes[]
-     * @return {Array<Occurrence>}
-     */
-    public static occurrenceMapToArray (occurrenceMap : Map<string, Array<number>>) : Array<Occurrence> {
-        let occurrences : Array<Occurrence> = [];
-        for (let [term, indexes] of occurrenceMap.entries()) {
-            let occurrence = new Occurrence(term, indexes);
-            occurrences.push(occurrence);
+    public static occArrayToMap(arr : Array<Occurrence>) : Map<string, [string, Array<number>]> {
+        let map : Map<string, [string, Array<number>]> = new Map();
+        for (let occ of arr) {
+            map.set(occ.term.term, [occ.term.id, occ.positions]);
```

Comment:
Well, that's a matter of intended use case. I think it would be a lot more sensible to be able to query by ID.